### ISSUE TITLE
Convert all Kubernetes v2 stages to React

### ISSUE THREAD

**Issue by maggieneterval** [spinnaker/spinnaker#4254]
As discussed in the UI SIG, many community members would like to contribute to Deck, but there is a high barrier to entry based on the confusing combination of AngularJS and React. Converting all of the Kubernetes V2 provider stages to React will increase future development velocity, allow for more component-level testing, and hopefully unlock more community-contributed bug fixes and feature work. Once the conversion of the stages is complete, I will write a separate issue for converting the rest of the Kubernetes v2 provider to React.

### Goals
- Scope out which stages and dependent components need to be converted to React.
- Track progress of the conversion and note any issues or useful patterns to hopefully help with the conversion of the other modules, perhaps adding more best-practices documentation to Deck’s wiki.

### Stages

- [ ] Delete Manifest stage
  - [ ] Replace all uses of `stageFieldUpdated` with `updateStageField`
- [ ] Deploy Manifest stage
  - [ ] Convert `deployManifestConfig.html` and `deployManifestConfig.controller.ts` to new `DeployManifestConfig.tsx` component.
  - [ ] Remove `/manifest/wizard/basicSettings.component.ts` and replace with `ManifestBasicSettings.tsx`.
- [ ] Disable Manifest stage
  - [ ] Replace all uses of `stageFieldUpdated` with `updateStageField`
- [ ] Enable Manifest stage
  - [ ] Replace all uses of `stageFieldUpdated` with `updateStageField`
- [ ] Find Artifacts From Resource Stage
  - [ ] Convert `findArtifactsFromResourceConfig.html` and `findArtifactsFromResourceConfig.controller.ts` to new `FindArtifactsFromResourceConfig.tsx` component.
- [ ] Patch Manifest stage
  - [ ] Convert `patchManifestConfig.html` and `patchManifestConfig.controller.ts` to new `PatchManifestConfig.tsx` component.
  - [ ] Convert `patchOptionsForm.component.ts` to new `PatchOptionsForm.tsx` component.
- [ ] Scale Manifest stage
  - [ ] Convert `scaleManifestConfig.html` and `scaleManifestConfig.controller.ts` to new `ScaleManifestConfig.tsx` component.
  - [ ] Convert `scaleSettingsForm.component.ts` to new `ScaleSettingsForm.tsx` component. Note that this is also used in `/manifest/scale/scale.html` for the ad-hoc scale form, so will need to be portable via `react2Angular` for this component until it is also converted to React.
- [ ] Undo Rollout Manifest stage
  - [ ] Convert `undoRolloutManifestConfig.html` and `undoRolloutManifestConfig.controller.ts` to new `UndoRolloutManifestConfig.tsx` component.

**Comment by maggieneterval** [comment]
Just a note that we should update each stage to use the new `FormikStageConfig` component rather than calling `updateStageField`. I am currently working on the Deploy Manifest stage but the other stages are still up for grabs :)

**Comment by spinnakerbot** [comment]
This issue hasn't been updated in 45 days, so we are tagging it as 'stale'. If you want to remove this label, comment:

> @spinnakerbot remove-label stale

**Comment by maggieneterval** [comment]
I'm moving this to the backlog for now because I'm not actively working on it -- if you would like to work on this, please feel free to reach out to me!

**Comment by rumit-opsmx** [comment]
@maggieneterval , @moki-daniel is the issue resolved, If not I would like to work on this.

**Comment by moki-daniel** [comment]
not yet closed @rumit-opsmx, feel free to resolve kindly
